flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointOptions.java

```diff
@@ -47,29 +50,56 @@
 
 	private final boolean isUnalignedCheckpoint;
 
+	private final long alignmentTimeout;
```

Comment:
No, `alignmentTimeout` is a global timeout, measured against the `CheckpointBarrier#getTimestamp`, so since the creation of the `CheckpointBarrier` on the `CheckpointCoordinator`. If `CheckpointCoordinator` has timed out on an upstream task, it will be timed out for any downstream operator as well.